example/mbed-os-5/mbed-client-cli.lib

```diff
@@ -1 +1 @@
-https://github.com/ARMmbed/mbed-client-cli
+https://github.com/ARMmbed/mbed-client-cli#4c44e6c98f5c01a578477d73ebfa7b6ed87c1fed
```

Comment:
Using hardcoded version like this doesn't allow to test same commit than what is cloned - not good solution either.
Propose:
1) (git)remove this mbed-client-cli.lib 
2) (Jenkinsfile) move example folder out of git clone folder
3) (Jenkinsfile) generate mbed-client-cli.lib file based on commit_id (`https://github.com/armmbed/mbed-client-cli#<commit_id>`) using CI env variable
4) (Jenkinsfile) compile example

or

do this hacky solution for now and wait until mbed-cli is fixed..

Reply:
if mbed-cli fixes needs long time to deliver, this hacky solution is accpetable, since testing mbed-os5 example and compile binary is imortant test.

Reply:
Let's merge this for now to avoid blocking other PR's